packages/internal/src/validateSchema.ts

```diff
@@ -6,20 +6,76 @@ import { DocumentNode, Kind, ObjectTypeDefinitionNode, visit } from 'graphql'
 import { rootSchema } from '@redwoodjs/graphql-server'
 import { getPaths } from '@redwoodjs/project-config'
 
+import { isServerFileSetup, isRealtimeSetup } from './project'
+
 export const DIRECTIVE_REQUIRED_ERROR_MESSAGE =
   'You must specify one of @requireAuth, @skipAuth or a custom directive'
 
 export const DIRECTIVE_INVALID_ROLE_TYPES_ERROR_MESSAGE =
   'Please check that the requireAuth roles is a string or an array of strings.'
-export function validateSchemaForDirectives(
+
+export const RESERVED_TYPES = [
+  'Int',
+  'Float',
+  'Boolean',
+  'String',
+  'DateTime',
```

Comment:
Not sure how helpful it is to mitigate these kinds of issues, but the Prisma IDE actually warns on these (ID, uid, as, etc. are ok; just these ones error out):

<img width="1532" alt="image" src="https://github.com/redwoodjs/redwood/assets/32992335/18a2b7cd-ab30-4c53-8464-87774be46edc">

Reply:
Right. Prisma will warn, but you could write a type SDL that is not Prisma backed and for some reason create a GraphQL schema type called "Float". This validation checks the GraphQL not database schema.

Your SDL could be for a third-party API for buying fishing floats :)

It's unlikely, but since these are "reserved names" (typically scalars even in the case for AWS Amplify which is GraphQL https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html) it seems prudent to prevent/discourage use.

Note that this doesn't 100% cover the case where types can collide -- if say you use a custom scalar like Currency and have type already called Currency.

Reply:
Leaving these in for now.